crates/notion-core/src/inventory/mod.rs

```diff
@@ -108,23 +117,9 @@ impl Inventory {
         Ok(Inventory {
             node: NodeCollection::load()?,
             yarn: YarnCollection::load()?,
+            packages: PackageCollection::load()?,
         })
     }
-
-    /// Fetches a Tool version matching the specified semantic versioning requirements.
-    pub fn fetch(
```

Comment:
Likely beyond the scope of this PR, but it seems like we are giving up a lot of the simplification gains that were made when consolidating all of the `fetch` methods into a single function. I see that the `DistroVersion` type is gone, so perhaps it's worth creating an issue to revisit if it's possible to re-combine these methods.

Reply:
I re-opened #87 for this, which I had previously closed. It may be possible to re-combine these methods, but it wasn't obvious at the time, because `PackageVersion` was too different from `Version` and `NodeVersion`, and there were some other issues but I forget what exactly at the moment.